src/julia_threads.h

```diff
@@ -74,6 +77,17 @@ typedef struct _jl_tls_states_t {
 #  define JL_CPU_WAKE_NOOP 1
 #endif
 
+// Copied from libuv. Add `JL_CONST_FUNC` so that the compiler
+// can optimize this better.
+static inline unsigned long JL_CONST_FUNC jl_thread_self(void)
+{
+#ifdef _OS_WINDOWS_
+    return (unsigned long)GetCurrentThreadId();
+#else
+    return (unsigned long)pthread_self();
+#endif
+}
```

Comment:
The way I see it is that there's basically no way we can get rid of all the platform dependent code and using libuv for everything. For complex stuff like io and event loops libuv has a nice abstraction and we can use it but for simple or things that doesn't exist in libuv (especially threading and signal handling) I'm in favor of just implement it ourselves. Of course if we can get more features upstream in the way we want it's great but otherwise maintaining our own shouldn't be a big problem.